01_classify_stratified/nbconverted/expression_vs_methylation.py

```diff
@@ -0,0 +1,254 @@
+
+sns.set({'figure.figsize': (16, 6)})
+sns.set_style('whitegrid')
+fig, axarr = plt.subplots(1, 2)
+
+# all plots should have the same axes for a fair comparison
+xlim = (-0.75, 0.75)
+# TODO: maybe adjust these afterward?
+y_max = all_results_df.nlog10_p.max()
+ylim = (0, y_max+3)
+
+data_types = sorted([dt for dt in all_results_df.training_data.unique() if dt != 'expression'])
+exp_results_df = results_df[results_df.training_data == 'expression']
+
+for ix, training_data in enumerate(data_types):
+    ax = axarr[ix]
+    data_results_df = results_df[results_df.training_data == training_data]
+    compare_results_df = au.compare_results(exp_results_df,
+                                            pancancer_df=data_results_df,
+                                            identifier='identifier',
+                                            metric='aupr',
+                                            correction=True,
+                                            correction_method='fdr_bh',
+                                            correction_alpha=SIG_ALPHA,
+                                            verbose=True)
+    compare_results_df.rename(columns={'identifier': 'gene'}, inplace=True)
+    compare_results_df['nlog10_p'] = -np.log10(compare_results_df.corr_pval)
+    sns.scatterplot(data=compare_results_df, x='delta_mean', y='nlog10_p', hue='reject_null',
+                    hue_order=[False, True], ax=ax)
+
+    # add vertical line at 0
+    ax.axvline(x=0, linestyle='--', linewidth=1.25, color='black')
+    # add horizontal line at statistical significance threshold
+    l = ax.axhline(y=-np.log10(SIG_ALPHA), linestyle='--', linewidth=1.25, zorder=-1)
+    # label horizontal line with significance threshold
+    # (matplotlib makes this fairly difficult, sadly)
+    ax.text(0.5, -np.log10(SIG_ALPHA)+0.1,
+            r'$\mathbf{{\alpha = {}}}$'.format(SIG_ALPHA),
+            va='center', ha='center', color=l.get_color(),
+            backgroundcolor=ax.get_facecolor(),
+            zorder=0)
+    ax.set_xlabel('AUPR({}) - AUPR(expression)'.format(training_data), size=14)
+    ax.set_ylabel(r'$-\log_{10}($adjusted $p$-value$)$', size=14)
+    ax.set_xlim(xlim)
+    ax.set_ylim(ylim)
+    ax.legend(title=r'Reject $H_0$', loc='upper left')
+    ax.set_title(r'Mutation prediction, expression vs. {}'.format(training_data), size=14)
+
+    text_labels = label_points(compare_results_df['delta_mean'],
+                               compare_results_df['nlog10_p'],
+                               compare_results_df.gene,
+                               ax)
+    adjust_text(text_labels,
+                ax=ax,
+                expand_text=(1., 1.),
+                lim=5)
+    
+    print('{}: {}/{}'.format(training_data,
+                             np.count_nonzero(compare_results_df.reject_null),
+                             compare_results_df.shape[0]))
+
+
+# In[9]:
+
+
+all_results_df[all_results_df.gene == 'FBXW7']
```

Comment:
Is there a reason you pull out FBXW7 at the end?

Reply:
I was looking at its results in detail since it's one of the few genes in our gene set involved in DNA damage repair (see, e.g. [this paper](https://doi.org/10.1016/j.molcel.2015.12.010)), in addition to BRCA1/2 and ATM. I wanted to look at these genes specifically to see if their prediction based on mutational signatures was somewhere close to significance, or just not working at all. Results are mixed.

I think I can just remove this line now, though. Good catch.